mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToHeader.java

```diff
@@ -0,0 +1,130 @@
+ *     </code>
+ * </pre>
+ */
+public class ICALToHeader extends GenericMailet {
+    public static final String ATTRIBUTE_PROPERTY = "attribute";
+    public static final String ATTRIBUTE_DEFAULT_NAME = "icalendar";
+
+    public static final String X_MEETING_UID_HEADER = "X-MEETING-UID";
+    public static final String X_MEETING_METHOD_HEADER = "X-MEETING-METHOD";
+    public static final String X_MEETING_RECURRENCE_ID_HEADER = "X-MEETING-RECURRENCE-ID";
+    public static final String X_MEETING_SEQUENCE_HEADER = "X-MEETING-SEQUENCE";
+    public static final String X_MEETING_DTSTAMP_HEADER = "X-MEETING-DTSTAMP";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ICALToHeader.class);
+
+    private String attribute;
+
+    @Override
+    public String getMailetInfo() {
+        return "ICALToHeader Mailet";
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        attribute = getInitParameter(ATTRIBUTE_PROPERTY, ATTRIBUTE_DEFAULT_NAME);
+        if (Strings.isNullOrEmpty(attribute)) {
+            throw new MessagingException("attribute can not be empty or null");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        if (mail.getAttribute(attribute) == null) {
+            return;
+        }
+        try {
+            Map<String, Calendar> calendars = (Map<String, Calendar>) mail.getAttribute(attribute);
+            Optional<Calendar> calendar = calendars.values()
+                .stream()
+                .findAny();
+            if (calendar.isPresent()) {
+                writeToHeaders(calendar.get(), mail.getMessage());
+            }
+        } catch (ClassCastException e) {
+            LOGGER.error("Received a mail with {} not being an ICAL object for mail {}", e, attribute, mail.getName());
+        }
+    }
+
+    @VisibleForTesting
+    String getAttribute() {
+        return attribute;
+    }
+
+    private void writeToHeaders(Calendar calendar, MimeMessage mimeMessage) throws MessagingException {
+        VEvent vevent = (VEvent) calendar.getComponent("VEVENT");
+        addIfPresent(mimeMessage, X_MEETING_METHOD_HEADER, calendar.getMethod());
+        addIfPresent(mimeMessage, X_MEETING_UID_HEADER, vevent.getUid());
+        addIfPresent(mimeMessage, X_MEETING_RECURRENCE_ID_HEADER, vevent.getRecurrenceId());
+        addIfPresent(mimeMessage, X_MEETING_SEQUENCE_HEADER, vevent.getSequence());
+        addIfPresent(mimeMessage, X_MEETING_DTSTAMP_HEADER, vevent.getDateStamp());
+        mimeMessage.saveChanges();
+    }
+
+    private void addIfPresent(MimeMessage mimeMessage, String headerName, Property property) {
+        if (property != null) {
+            try {
+                mimeMessage.addHeader(headerName, property.getValue());
+            } catch (MessagingException e) {
+                LOGGER.warn("Could not add header {} with value {}", headerName, property.getValue());
```

Comment:
To explain this one a bit better, I really think centralizing mailet logging thew the MailetContainer (Akka log() calls of Generic mailets) will make fined grain configuration of logging (using class names for instance) will be wayharder as an admin.

Concerning the log method of the generic mailet, I don't see why I would have to use a poor Logging API, without placehorlder, and were you will have to guess the log level of what you are doing.

For older mailets, maybe we can not change it for now but for new mailets, maybe we should go away from this API